docs/concepts/macros/sqlmesh_macros.md

```diff
@@ -318,6 +328,21 @@ SELECT
 FROM table
 ```
 
+Another example is conditionally executing a pre/post-statement depending on the current [runtime stage](./macro_variables.md#predefined-variables). For instance, the following `@IF` pre-statement will only be executed at model creation time:
+
+```sql linenums="1"
+MODEL (
+  ...
+);
+
+@IF(
```

Comment:
have you tested this? i think you want this as a post statement and not a pre statement

Reply:
not yet, i can try to test this specific example though soon

Reply:
to be clear, i did test using both `create schema ...` and `create table ...` as pre-statements, seemed good

Reply:
yeah what you're saying makes sense, I'll try to refine the example

Reply:
updated the example to use an evaluation-time post-statement instead